Add website- and storeview-code in stores admin grid

aschrammel opened the pull request:
This PR adds the webiste- and storeview-code in the stores admin grid like in Magento 1 stores overview.

### Description
The PR joins additionally the `code` field from the `store` table to the admin grids row and adds the website code as well as the store view code in the respective renderers.

### Fixed Issues (if relevant)
-

### Manual testing scenarios
1. Go to "Stores" > "Settings / All Stores"
2. Before applying the PR there are only listed Website, Store and StoreView by their names
3. After applying the PR there are the respective codes shown beneath the Website and StoreView names

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

Commit:
:sparkles: Add website- and storeview-code

Add website- and storeview-code in stores Admin-Grid.

mzeis commented:
Hi @aschrammel,

thank you very much for your contribution! Would you consider adding the store name, too?

aschrammel commented:
Hi @mzeis,

thanks for your feedback. 
I think you mean the store code, as the name is already shown - am i right?

mzeis commented:
@aschrammel Exactly :)

Commit:
:ok_hand: Add store code

Add store code in store admin grid.

mzeis commented:
@magento-engcom-team give me new test instance

magento-engcom-team commented:
Hi @mzeis, here is your [new test Magento instance](http://pr-14156.engcom.dev.magento.com)

mzeis commented on app/code/Magento/Backend/Block/System/Store/Grid/Render/Group.php:
```diff
@@ -27,6 +27,7 @@ public function render(\Magento\Framework\DataObject $row)
         $this->escapeHtml($row->getData($this->getColumn()->getIndex())) .
-        '</a>';
+        '</a><br />'
+        . '(Code: ' . $row->getGroupCode() . ')';
```
@aschrammel Please can you make "Code" translatable in all 3 places (like "Edit Store" here: [Group.php](https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/Backend/Block/System/Store/Grid/Render/Group.php#L23-L25) and [en_US.csv](https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/Backend/i18n/en_US.csv)?

Then we should be good to go. Thanks!

Commit:
:ok_hand: :globe_with_meridians: Make "Code" translatable

Make "Code" string translateable. Translation for "Code" is already
contained in i18n/en_US.csv (Line 101).

Commit:
:twisted_rightwards_arrows: Merge branch '2.2-develop' of github.com:magento/magento2 into 2.2-develop

aschrammel commented:
@mzeis Code is translatable now 🌐

mzeis commented:
@magento-engcom-team give me new test instance

magento-engcom-team commented:
Hi @mzeis, here is your [new test Magento instance](http://pr-14156.engcom.dev.magento.com)

mzeis commented:
@aschrammel Thanks for you improvements, looking good! 👍 I'm handing over to QA & merging.

magento-engcom-team commented:
Hi @mzeis, thank you for the review.
[ENGCOM-968](https://jira.corp.magento.com/browse/ENGCOM-968) has been created to process this Pull Request

aschrammel commented:
@magento-engcom-team thanks for acceptance 😄
In #14157 is the respective forward port to 2.3 if wanted.

mzeis commented:
Thanks! I'll process it once the merge for this one is completed :)

mzeis commented:
Hi @aschrammel, thanks again for your PR! It has been merged into `2.2-develop`. We now can proceed with #14157.
